functions/check_version_and_known_hosts.pp

```diff
@@ -0,0 +1,13 @@
+function peadm::check_version_and_known_hosts(
+  String $target_version,
+  Optional[Peadm::Known_hosts]      $r10k_known_hosts         = undef,
+) {
+  $version_check = SemVer($target_version) >= SemVer('2023.3.0')
+
+  out::message("Version check: ${version_check}")
+
+  # If the version is 2023.3 or greater and known_hosts is undef, print a message
+  if ($version_check and $r10k_known_hosts == undef) {
+    out::message("URGENT UPGRADE NOTICE:\nWhen you are upgrading to PE 2023.3 or later, it's crucial to ensure that the 'known_hosts' parameter for R10k is populated. Starting from PE 2023.3, Code Manager will not operate correctly without this. For more information, see https://puppet.com/docs/pe/2023.3/code_mgr_config.html#known-hosts-parameter")
```

Comment:
This warning will likely print for ever future upgrade unless we also test that $current_version < 2023.3.

Reply:
Also, for the warning message (and upgrade docs above, probably) $r10k_known_hosts is only needed if $r10k_remote is using the ssh protocol. If it's https or file, for example, it isn't required.